module/zfs/zio.c

```diff
@@ -1802,9 +1813,9 @@ zio_write_compress(zio_t *zio)
 			 * in that we charge for the padding used to fill out
 			 * the last sector.
 			 */
-			ASSERT3U(spa->spa_min_alloc, >=, SPA_MINBLOCKSHIFT);
-			size_t rounded = (size_t)roundup(psize,
-			    spa->spa_min_alloc);
+			ASSERT3U(spa->spa_gcd_alloc, >=, SPA_MINBLOCKSIZE);
```

Comment:
@amotin - could you please check this line? The ASSERT should be for `SPA_MINBLOCKSIZE` instead of `SPA_MINBLOCKSHIFT` if I'm not missing anything.

Reply:
Thank you, I have removed it.